### ISSUE TITLE
EditorConfig

### ISSUE THREAD

**Issue by toastal** [purescm/purescm#219]
I would like to add an [EditorConfig](https://editorconfig.org/) file to the project since I use tabs by default (for my own a11y), which doesn’t work  with how this project is set up. Also there appears to be some small inconsistencies with whitespace in the project as it exists (ignoring `vendor` + `legacy` obviously). I would prefer then to also add checking to the CI pipeline, but to do so I would also prefer a bump to `nixpkgs` to get access to `lib.fileset` to be efficient about said check, but the current lock is to a version that is almost a year old now.

**Comment by f-f** [comment]
We are happy with bumping nixpkgs to some stable ref (e.g. 23.11)

The whole project should be formatted with `purs-tidy`, and if not then we should fix that. Other than that, what inconsistencies are there?

Re editor config: isn't this a user-specific setting? I wouldn't want to commit a settings file that defaults to tabs

**Comment by f-f** [comment]
Ah I see - I think we can get an `.editorconfig` then, and go with option 3 for dealing with the indent_size issue.

In the case of the Scheme files in the `test-snapshot` folder, they are generated by the backend, so we'd need to fix the pretty-printer to format properly.

**Comment by anttih** [comment]
Those `test-snapshot/` files are input files which are formatted by hand. You are probably not running the check for the _output_ files.

**Comment by anttih** [comment]
I was just referring to the output of `editorconfig-checker` that you pasted as it contained only snapshot input files.
